packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -182,9 +179,12 @@ const DropdownMenuTrigger = React.forwardRef<DropdownMenuTriggerElement, Dropdow
             }
           })}
           onKeyDown={composeEventHandlers(props.onKeyDown, (event) => {
-            if (!disabled && event.key === 'ArrowDown') {
-              event.preventDefault();
+            if (disabled) return;
+            if (['Enter', ' '].includes(event.key)) context.onOpenToggle();
+            if (event.key === 'ArrowDown') {
               context.onOpenChange(true);
+              // prevent arrow keypress from scrolling window
+              event.preventDefault();
```

Comment:
I think we need this for space as well as it looks like the page scrolls when using space in non-modal case (must be similar reason to what I was showing you in `Select`).